### ISSUE TITLE
Unable to install, Error: Call to a member function documentationConfig() on string

### ISSUE THREAD

**Issue by Jason6809** [DarkaOnLine/L5-Swagger#415]
- L5-Swagger Version: 8.0.9
- PHP Version (`php -v`): 7.3.29
- OS: Mac OS Big Sur 11.6

### Description:
Call to a member function documentationConfig() on string

Terminal output:
  ```
Error 

  Call to a member function documentationConfig() on string

  at vendor/darkaonline/l5-swagger/src/routes.php:14
     10▕ 
     11▕     $documentations = config('l5-swagger.documentations', []);
     12▕ 
     13▕     foreach (array_keys($documentations) as $name) {
  ➜  14▕         $config = $configFactory->documentationConfig($name);
     15▕ 
     16▕         if (! isset($config['routes'])) {
     17▕             continue;
     18▕         }

      +13 vendor frames 
  14  [internal]:0
      Illuminate\Foundation\Application::Illuminate\Foundation\{closure}(Object(L5Swagger\L5SwaggerServiceProvider))

      +5 vendor frames 
  20  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1
```

This package working perfectly fine on my another project a few week ago, but it suddenly bugged on my new project today. I have tried commands, `php artisan config:cache` and `route:cache`, `composer update` and `dumpautoload`, still can't fix it .

### Steps To Reproduce:
just run this command: composer require "darkaonline/l5-swagger"

**Comment by DarkaOnLine** [comment]
Hi, Have you solved this? This is really strange issue 🤔

**Comment by damienorny** [comment]
Hi there 👋
This happened to us recently and since my research got me on this issue I thought I would give you some insights. It was due to a change in the autoload order. Here is what I can give you to help debugging : 
- We used `hoa/mime` dep to get mimetype from filename
- `hoa/mime` and Laravel had conflicts on the `resolve()` method
- We had to modify our `bootstrap/autoload.php` file like this : 
```php
//we need to load helpers first to avoid laravel resolve() function override by hoa/mime resolve() function
require __DIR__.'/../vendor/laravel/framework/src/Illuminate/Foundation/helpers.php';
require __DIR__.'/../vendor/autoload.php';
```
- Removing a random package (`alecrabbit/php-cli-snake` in our case) caused an autoload order change that made `/hoa/protocol/Wrapper.php` being loaded before `/laravel/framework/src/Illuminate/Foundation/helpers.php` and then causing the same `Call to a member function documentationConfig() on string` error

As a fix we got rid of `hoa/mime` and used a simpler piece of code for file type detection + we got rid of the additional line in the `bootstrap/autoload.php` file.

So my guess is that you have another dependency messing with the `resolve` function @Jason6809 

Hope this helped :)
